transport/http/outbound.go

```diff
@@ -339,7 +349,27 @@ func (o *Outbound) getPeerForRequest(ctx context.Context, treq *transport.Reques
 
 func (o *Outbound) createRequest(treq *transport.Request) (*http.Request, error) {
 	newURL := *o.urlTemplate
-	return http.NewRequest("POST", newURL.String(), treq.Body)
+	hreq, err := http.NewRequest("POST", newURL.String(), treq.Body)
+	if err != nil {
+		return nil, err
+	}
+	// grpc in, http out case
+	if treq.Transport == "grpc" {
+		// from https://tools.ietf.org/html/rfc7540#section-8.1.2.3
+		// For ":authority", ...
+		// An intermediary that converts an HTTP/2 request to HTTP/1.1 MUST
+		// create a Host header field if one is not present in a request by
+		// copying the value of the ":authority" pseudo-header field.
+		if v, ok := treq.Headers.Get(http2AuthorityPseudoHeader); ok && hreq.Host == "" {
+			hreq.Host = v
+		}
+		// strip all http2 pseudo-header fields
+		for _, k := range http2PseudoHeaders {
+			treq.Headers.Del(k)
+		}
+	}
```

Comment:
What do you think about putting this logic into a function `stripHTTP2HeadersIfNeeded()`